plugins/org.obeonetwork.m2doc.doc/doc/user/M2Doc User Guide.html

```diff
@@ -173,6 +173,18 @@ <h4 id="Siriustableproviders">Sirius table providers</h4>
 			<b>Warning</b>: do not forget the single quotes ' around the string in an AQL Expression: 
 			<code>title:"'Table name'"</code>. The description ID does not need such quotes because it is of type String.
 		</p>
+		<h4 id="UpdatestrategyforSiriusRepresentation">Update strategy for Sirius Representation</h4>
+		<p>In the cases of generating a diagram with {m:diagram} or a table with {m:wtable}, the M2Doc generation can take place on an unsynchronized representation (diagram or table). Consequently, a parameter has been introduced in the genConf model to allow users to specify whether representations must be refreshed before the M2Doc generation.</p>
+		<p>
+			<img border="0" src="images/image019.png"/>
+		</p>
+		<p>The boolean parameter &#171;refreshRepresentation&#187; can be used to define following update strategy : 
+			<br/>- if it is set to false, the M2Doc generation take the state of graphical model without update it
+			<br/>- if it is set to true, the M2Doc generation take the updated with semantic model and arranged model like if we open diagram or table in Sirius representation view.
+			<br/>By default, the parameter is set to false.
+		</p>
+		<p>Be careful, enabling the automatic update of representations can be time-consuming and could have a high impact on the generation time, especially on large models.</p>
+		<p>In case of enabling the automatic update of representations, if SIRIUS view is not open, the update and arrange effect is automatically save on graphical model. On the other hand, if SIRIUS view is open, the arrange and update is done by view. Your representation was dirty until you save your graphical diagram and the M2Doc Generation is done with update whatever the value of the &#171;RefreshRepresentation&#187; parameter.</p>
```

Comment:
When the automatic refresh of representations is on, generating a document may modify your model:
<ul>
<li>If the session is dirty (currently has unsaved changes), then the session will still be dirty after the generation, but it may contain more changes than before the generation (all the generated representations will have been refreshed)</li>
<li>If the session is clean (no unsaved changes), then the session will still be clean after the generation, but it may have been modified and saved during the generation (all the generated representations will have been refreshed)</li>
</ul>

*I think this proposal is easier to understand, but is it correct?*